File: PetalBid.Api/Domain/Entities/Product.cs
namespace PetalBid.Api.Domain.Entities;

public class Product
{
	public int Id { get; set; }
	public string Name { get; set; } = string.Empty;
	public double Weight { get; set; }
	public string ImageBase64 { get; set; } = string.Empty;
	public string Species { get; set; } = string.Empty;
	public double? PotSize { get; set; }
	public double? StemLength { get; set; }
	public int Stock { get; set; }
	public double MinimumPrice { get; set; }

	// Link to Supplier
	public int SupplierId { get; set; }
	public Supplier Supplier { get; set; } = null!;

	// AuctionId is now nullable
	public int? AuctionId { get; set; }
	public Auction? Auction { get; set; }

	// Maximum price per unit that a buyer can bid for the product in an auction
	public double? MaxPricePerUnit { get; set; }

	// Date when the product is available for sale
	public DateTime? SaleDate { get; set; }
}

File: client-app/src/types/product.ts
/**
 * Represents a product or lot offered by a supplier for auction.
 */
export interface Product {
	/** The unique identifier for the product. */
	id: number;
	/** The common name of the product (e.g., 'Red Roses'). */
	name: string;
	/** The weight of the product, typically in kilograms. */
	weight: number;
	/** A Base64 string of the processed AVIF image. */
	imageBase64: string;
	/** The scientific or common species name (e.g., 'Rosa'). */
	species: string;
	/** The diameter of the pot in centimeters, if applicable. */
	potSize?: number;
	/** The length of the stem in centimeters, if applicable. */
	stemLength?: number;
	/** The total quantity available in the lot. */
	stock: number;
	/** The minimum price per unit that the supplier will accept. */
	minimumPrice: number;
	/** The ID of the supplier who owns this product. */
	supplierId: number;
	/** The ID of the auction this product is assigned to, if any. */
	auctionId?: number;
	/** The maximum price per unit that a buyer can bid for the product in an auction. */
	maxPricePerUnit?: number;
}
